### PR TITLE
[8407] wm monitor main implementation

### LINKED ISSUE #8407: Monitor module implementation of wm_context functions
|Branch|
|---|
|[8407-wm-main-destroy-impl](https://github.com/wazuh/wazuh/tree/8407-wm-main-destroy-impl)|

## Description
This issue points to the unification of processes on the manager side, in which it aims to have less processes.

One of the biggest challenges here is to decompose the main of the old module, separating what is the initialization, from the main processing loop.

For this type of modification, the main thing is to normalize the module, to maintain the structure and design of wazuh-modulesd, as it is used in the pre-existing modules.

## DoD
- [x] Expose methods (Mainloop, shutdown).
- [x] Initialize module and call the main loop.
- [x] Initialize argument received in wmodule calls. --  context->data
- [x] Functional tests - In progress.
- [x] PR CI Pass (Only Build and unit-tests)

### PR DESCRIPTION
**Pull request by tinachoB**
|Related issue|
|---|
|Closes #8407 |

## Description
This issue points to the unification of processes on the manager side, in which it aims to have less processes.

One of the biggest challenges here is to decompose the main of the old module, separating what is the initialization, from the main processing loop.

For this type of modification, the main thing is to normalize the module, to maintain the structure and design of wazuh-modulesd, as it is used in the pre-existing modules.

## DoD
- [x] Expose methods (Mainloop, shutdown).
- [x] Initialize module and call the main loop.
- [x] Initialize argument received in wmodule calls.
- [X] Functional tests
- [ ] CI Pass (Only Build and unit-tests)

### CODE REVIEW

File: src/config/wmodules-monitor.c [hunk 1]
```diff
@@ -10,14 +10,17 @@
...
-#include "wazuh_modules/wmodules.h"
+#include "../wazuh_modules/wmodules.h"
 #include "../monitord/monitord.h"
 
 // Parse XML configuration
-int wm_monitor_read(const OS_XML *xml, XML_NODE node, wmodule *module) {
+int wm_monitor_read(__attribute__((unused)) const OS_XML *xml, __attribute__((unused)) XML_NODE node, wmodule *module) {
     if (module) {
         module->context = &WM_MONITOR_CONTEXT;
         module->tag = strdup(module->context->name);
```
Comment (konangc87) [inline src/config/wmodules-monitor.c:20]: this should be os_strdup to avoid coverity issues.

File: src/wazuh_modules/wm_monitor.c [hunk 2]
```diff
@@ -51,19 +51,91 @@ void* wm_monitor_main(wm_monitor_t *data) {
...
+    switch (w_is_worker()){
+        case 0:
+            worker_node = false;
+            break;
+        case 1:
+            mdebug1("Cluster client node: Disabled the agent monitoring");
+            worker_node = true;
+            mond.monitor_agents = 0;
+            break;
+    }
```
Comment (konangc87) [inline src/wazuh_modules/wm_monitor.c:127]: maybe you could use
if(w_is_worker() == 1) {
    ....
} else {
    worker_node = false;
}

**Review by konangc87 — CHANGES_REQUESTED** [review]

File: src/config/wmodules-monitor.c (hunk 1 above)
Comment (tinachoB) [reply src/config/wmodules-monitor.c:20]: Agree! I'll change it.

File: src/wazuh_modules/wm_monitor.c (hunk 2 above)
Comment (tinachoB) [reply src/wazuh_modules/wm_monitor.c:127]: will do

**Review by konangc87 — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**